dlog/commitment/src/srs.rs

```diff
@@ -70,6 +76,38 @@ where
         self.g.len()
     }
 
+    /// Compute commitments to the lagrange basis corresponding to the given domain and
+    /// cache them in the SRS
+    pub fn add_lagrange_basis(&mut self, domain: D<G::ScalarField>) {
+        let n = domain.size();
+        if n > self.g.len() {
+            panic!("add_lagrange_basis: Domain size {} larger than SRS size {}", n, self.g.len());
+        }
+
+        if self.lagrange_bases.contains_key(&n) {
+            return;
+        }
+
+        let mut lg: Vec<<G as AffineCurve>::Projective> =
+            self.g[0..n].iter().map(|g| g.into_projective()).collect();
+
+        group_fft::<<G as AffineCurve>::Projective>(
+            lg.as_mut_slice(),
+            domain.group_gen_inv,
+            domain.log_size_of_group,
+        );
+
+        let n_inv = <G::ScalarField as From<u64>>::from(n as u64)
```

Comment:
can't you write `let: G::ScalarField = (n as u64).inverse().unwrap().into();` ?

Reply:
What is inverse as u64?

Reply:
oops, 
```
let n: G::ScalarField = (n as u64).into();
let n_inv = n.inverse().except("n was 0");
```